src/main/groovy/netflix/nebula/dependency/recommender/provider/RecommendationProviderContainer.java

```diff
@@ -18,23 +18,25 @@
 import groovy.lang.Closure;
 import netflix.nebula.dependency.recommender.DependencyRecommendationsPlugin;
 import netflix.nebula.dependency.recommender.RecommendationStrategies;
-import org.gradle.api.Action;
-import org.gradle.api.GradleException;
-import org.gradle.api.Namer;
-import org.gradle.api.Project;
+import org.gradle.api.*;
 import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.internal.ConfigureByMapAction;
 import org.gradle.api.internal.DefaultNamedDomainObjectList;
+import org.gradle.api.model.ObjectFactory;
 import org.gradle.util.ConfigureUtil;
+import org.gradle.util.GradleVersion;
 
 import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 import java.util.*;
 
 import static netflix.nebula.dependency.recommender.DependencyRecommendationsPlugin.CORE_BOM_SUPPORT_ENABLED;
 
-public class RecommendationProviderContainer extends DefaultNamedDomainObjectList<RecommendationProvider> {
```

Comment:
Change from directly extending this class to using field with a public interface gives us more options on how to initialize the class.